src/Agent.Sdk/Util/IOUtil.cs

```diff
@@ -223,6 +227,51 @@ public static void DeleteFile(string path)
             }
         }
 
+        public static async Task DeleteDirectoryWithRetry(string path, CancellationToken cancellationToken)
+        {
+            for (int i = 0; i < MAX_RETRY_DELETION; i++)
+            {
+                try
+                {
+                    DeleteDirectory(path, cancellationToken);
+                    return;
+                }
+                // There is no reason to retry on DirectoryNotFoundException, SecruityException and UnauthorizedAccessException
+                catch (IOException)
```

Comment:
We should really have telemetry on retries with the goals of:
1. Measuring effectiveness
2. Root-causing and driving down retries.  

What we'd want to see: # of retries, exception type.  Ideally on catch we'd dump a mini-rca to the diagnostic log (e.g. handle, filter drivers, etc)

Reply:
Agreed to add telemetry publisher with another PR